ucm2/codecs/rt715-sdca/init.conf

```diff
@@ -0,0 +1,9 @@
++# RT715-sdca (aka RT714) specific volume control settings
+
+BootSequence [
+	cset "name='rt714 FU0A Capture Switch' 0"
+	cset "name='rt714 ADC 22 Mux' 'DMIC3'"
+	cset "name='rt714 ADC 23 Mux' 'DMIC4'"
+	cset "name='rt714 FU02 Capture Switch' 1"
+	cset "name='rt714 FU02 Capture Volume' 124 124"
```

Comment:
Perhaps, the initialization for the second stereo pair is missing here? In the kernel patch (https://lore.kernel.org/alsa-devel/7d74544f-77f2-db01-db3d-d1d8a56d576d@linux.intel.com/T/#m9d8869261e4531f537336da6b81085c1567034a9) the four stereo controls are set:

```
+	unsigned int capture_reg_H[] = {RT715_SET_GAIN_MIC_ADC_H,
+		RT715_SET_GAIN_LINE_ADC_H, RT715_SET_GAIN_MIX_ADC_H,
+		RT715_SET_GAIN_MIX_ADC2_H};
+	unsigned int capture_reg_L[] = {RT715_SET_GAIN_MIC_ADC_L,
+		RT715_SET_GAIN_LINE_ADC_L, RT715_SET_GAIN_MIX_ADC_L,
+		RT715_SET_GAIN_MIX_ADC2_L};
```

The stereo controls are:

```
        SOC_DOUBLE_R_EXT_TLV("ADC 07 Capture Volume", RT715_SET_GAIN_MIC_ADC_H,
                        RT715_SET_GAIN_MIC_ADC_L, RT715_DIR_IN_SFT, 0x3f, 0,
                        rt715_set_amp_gain_get, rt715_set_amp_gain_put,
                        in_vol_tlv),
        SOC_DOUBLE_R_EXT_TLV("ADC 08 Capture Volume", RT715_SET_GAIN_LINE_ADC_H,
                        RT715_SET_GAIN_LINE_ADC_L, RT715_DIR_IN_SFT, 0x3f, 0,
                        rt715_set_amp_gain_get, rt715_set_amp_gain_put,
                        in_vol_tlv),
        SOC_DOUBLE_R_EXT_TLV("ADC 09 Capture Volume", RT715_SET_GAIN_MIX_ADC_H,
                        RT715_SET_GAIN_MIX_ADC_L, RT715_DIR_IN_SFT, 0x3f, 0,
                        rt715_set_amp_gain_get, rt715_set_amp_gain_put,
                        in_vol_tlv),
        SOC_DOUBLE_R_EXT_TLV("ADC 27 Capture Volume", RT715_SET_GAIN_MIX_ADC2_H,
                        RT715_SET_GAIN_MIX_ADC2_L, RT715_DIR_IN_SFT, 0x3f, 0,
                        rt715_set_amp_gain_get, rt715_set_amp_gain_put,
                        in_vol_tlv),
```

If you need only one control, we can use the hardware rt714 register controls now. My impression was that there are more microphones then 2 are connected... Perhaps, I need to study, how the FU controls are build for rt714/715 (they come from topology files?).

Reply:
Ok, you have rt715-scda.c only in your development repo. It is not upstreamed and the FU02 (and maybe FU0E/FU0C boost) controls are problematic in my eyes - it contains the stereo volume/switch mapping/coupling to multiple hardware registers/inputs.

I created https://github.com/thesofproject/linux/issues/2696 .